tracer/src/Datadog.Trace/RemoteConfigurationManagement/RcmSubscriptionManager.cs

```diff
@@ -156,10 +149,19 @@ public byte[] GetCapabilities()
     {
         // capabilitiesArray needs to be big endian
 #if NETCOREAPP
-        var capabilitiesArray = _capabilities.ToByteArray(true, true);
+        var capabilitiesArray = _capabilities.ToByteArray(isUnsigned: true, isBigEndian: true);
 #else
         var capabilitiesArray = _capabilities.ToByteArray();
         Array.Reverse(capabilitiesArray);
+
+        if (capabilitiesArray.Length > 1 && capabilitiesArray[0] == 0)
+        {
+            // HACK: .NET Framework BigInteger.ToByteArray() can add a 0x00 byte to distinguish
```

Comment:
nitpicky, but I don't think this is actually just `.NET Framework` and is more specifically `BigInteger.ToByteArray()` where the `BigInteger` has been created with one of those "dangerous" numbers (example `new BigInteger(1 << 7)` will always create two bytes when `.ToByteArray()` is called regardless of runtime)